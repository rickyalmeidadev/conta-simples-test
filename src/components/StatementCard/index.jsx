import React from 'react';

const StatementCard = ({ id, name, price, date }) => (
  <li data-testid={id}>
    <div>
      <h2>{name}</h2>
    </div>
    <div>
      <span>R$ {price}</span>
      <span>{date}</span>
    </div>
  </li>
);

export default StatementCard;
