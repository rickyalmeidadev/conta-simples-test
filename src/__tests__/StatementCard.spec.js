import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import StatementCard from '../components/StatementCard';

afterEach(cleanup);

describe('<StatementCard />', () => {
  const mockCard = {
    id: 1,
    name: 'Facebook Inc.',
    price: 39.8,
    date: '2020-05-20T09:40:31-0300',
    debit: true,
  };

  it('renders correctly', () => {
    const { asFragment } = render(
      <Router>
        <StatementCard {...mockCard} />
      </Router>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the cards', () => {
    const { getByText, getByTestId } = render(
      <Router>
        <StatementCard {...mockCard} />
      </Router>,
    );

    expect(getByTestId(String(mockCard.id))).toBeDefined();
    expect(getByText(mockCard.name)).toBeDefined();
    expect(getByText(`R$ ${mockCard.price.toFixed(2)}`)).toBeDefined();
    expect(getByText('20/05/2020')).toBeDefined();
  });
});
