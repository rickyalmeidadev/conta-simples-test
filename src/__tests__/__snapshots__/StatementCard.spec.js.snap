// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatementCard /> renders correctly 1`] = `
<DocumentFragment>
  <li
    class="transactions__item"
    data-testid="1"
  >
    <div>
      <h2
        class="transactions__title"
      >
        Facebook Inc.
      </h2>
    </div>
    <div
      class="transactions__info"
    >
      <span
        class="transactions__value transactions__value--debit"
      >
        R$ 39.80
      </span>
      <span
        class="transactions__date"
      >
        <time
          datetime="1589978431000"
        >
          20/05/2020
        </time>
      </span>
    </div>
  </li>
</DocumentFragment>
`;
